src/frontend/js/components/ReferralTable/Input.tsx

```diff
@@ -1,18 +1,199 @@
+import * as Sentry from '@sentry/react';
 import debounce from 'lodash-es/debounce';
+import { stringify } from 'query-string';
 import React, { Dispatch, SetStateAction, useCallback, useState } from 'react';
-import Autosuggest from 'react-autosuggest';
-import { defineMessages, useIntl } from 'react-intl';
+import Autosuggest, {
+  SuggestionsFetchRequestedParams,
+} from 'react-autosuggest';
+import {
+  defineMessages,
+  FormattedMessage,
+  MessageDescriptor,
+  useIntl,
+} from 'react-intl';
 
-import { FiltersDict, QueryAutosuggestProps, SuggestionSection } from './types';
+import { appData } from 'appData';
+import { getUserFullname } from 'utils/user';
+import {
+  FilterColumns,
+  FiltersDict,
+  QueryAutosuggestProps,
+  Suggestion,
+  SuggestionSection,
+} from './types';
 
 const messages = defineMessages({
   queryPlaceholder: {
     defaultMessage: 'Search in referrals',
     description: 'Placeholder for the query input in referral table.',
     id: 'components.ReferralTable.Input.queryPlaceholder',
   },
+  suggestionSectionAssignees: {
+    defaultMessage: 'Assignees',
+    description:
+      'Title for the suggestion section in the search autocomplete dropdown.',
+    id: 'components.ReferralTable.Input.suggestionSectionAssignees',
+  },
+  suggestionSectionTopics: {
+    defaultMessage: 'Topics',
+    description:
+      'Title for the suggestion section in the search autocomplete dropdown.',
+    id: 'components.ReferralTable.Input.suggestionSectionTopics',
+  },
+  suggestionSectionUnits: {
+    defaultMessage: 'Units',
+    description:
+      'Title for the suggestion section in the search autocomplete dropdown.',
+    id: 'components.ReferralTable.Input.suggestionSectionUnits',
+  },
+  suggestionSectionUsers: {
+    defaultMessage: 'Requesters',
+    description:
+      'Title for the suggestion section in the search autocomplete dropdown.',
+    id: 'components.ReferralTable.Input.suggestionSectionUsers',
+  },
 });
 
+const SUGGESTION_SECTIONS = [
+  {
+    column: FilterColumns.TOPIC,
+    kind: 'topiclites',
```

Comment:
"topiclites", "unitlites" and "userlites" are everywhere in the code. It should be a const somewhere.